hudi-spark-datasource/hudi-spark/src/test/java/org/apache/hudi/client/TestHoodieClientMultiWriter.java

```diff
@@ -1081,11 +1327,23 @@ private JavaRDD<WriteStatus> createCommitWithInserts(HoodieWriteConfig cfg, Spar
     return result;
   }
 
+  private static void startSchemaEvolutionTransaction(HoodieTableMetaClient metaClient, SparkRDDWriteClient client, String nextCommitTime2, HoodieTableType tableType) throws IOException {
+    String commitActionType = CommitUtils.getCommitActionType(WriteOperationType.UPSERT, tableType);
+    client.startCommitWithTime(nextCommitTime2, commitActionType);
+    client.preWrite(nextCommitTime2, WriteOperationType.UPSERT, client.createMetaClient(true));
+    HoodieInstant requested = metaClient.createNewInstant(HoodieInstant.State.REQUESTED, commitActionType, nextCommitTime2);
+    HoodieCommitMetadata metadata = new HoodieCommitMetadata();
+    metadata.setOperationType(WriteOperationType.UPSERT);
+    client.createMetaClient(true).getActiveTimeline().transitionRequestedToInflight(
+        requested, Option.of(metadata.toJsonString().getBytes(StandardCharsets.UTF_8)));
```

Comment:
The timeline in table version 8 (default) uses Avro to serialize the commit metadata.  Is that changed here or the test utils are outdated?